src/main/java/com/exadel/discount/controller/FavoriteController.java

```diff
@@ -0,0 +1,66 @@
+package com.exadel.discount.controller;
+
+import com.exadel.discount.dto.favorite.BaseFavoriteDto;
+import com.exadel.discount.dto.favorite.FavoriteDto;
+import com.exadel.discount.entity.Favorite;
+import com.exadel.discount.service.FavoriteService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/favorites")
+public class FavoriteController {
+    private final FavoriteService favoriteService;
+
+    @Autowired
+    public FavoriteController(FavoriteService favoriteService) {
+        this.favoriteService = favoriteService;
+    }
+
+    @PostMapping("{userId}")
+    public ResponseEntity<FavoriteDto> addFavorite(@PathVariable final UUID userId,
+                                                     final BaseFavoriteDto baseFavoriteDto) {
+        Favorite favorite = favoriteService.addFavoriteToUser(userId, Favorite.from(baseFavoriteDto));
+        return new ResponseEntity<>(FavoriteDto.from(favorite), HttpStatus.OK);
+    }
+
+    @GetMapping
+    public ResponseEntity<List<FavoriteDto>> getAllFavorites() {
+        List<Favorite> allFavorites = favoriteService.findAllFavorites();
+        List<FavoriteDto> allFavoritesDto = allFavorites.stream()
+                .map(FavoriteDto::from)
+                .collect(Collectors.toList());
+        return new ResponseEntity<>(allFavoritesDto, HttpStatus.OK);
+    }
```

Comment:
We don't need to use ResponseEntity everywhere. If you choose http status 200, it can be just List<FavoriteDto>. And also mapping to dto should be at service layer